bson-codec-design/DESIGN.md

```diff
@@ -73,27 +81,29 @@ The `Registry` type will also have a generic codec registry, which only contains
 structs and one for maps. These codecs are only used if both the `TypeRegistry` and the
 `InterfaceRegistry` do not return a codec. The user can override these generic codecs.
 
+The `Registry` type is the object users are expected to pass around and use. Call sites are expected
+to acquire from a pool or create an `Encoder` or `Decoder`, use it, and then either return it to the
```

Comment:
A pool of registries?

Reply:
"acquire an `Encoder` or `Decoder` from a pool or create one"